### PR TITLE
Fixxes an issue which caused medics getting kicked for spamming

### PR DESCRIPTION
**Pull request by Dimiikou**
https://github.com/paulzhng/UCUtils/issues/4

Fixxes this issue

### CODE REVIEW

**Review by paulzhng — CHANGES_REQUESTED** [review]
In general, this idea is a good one. I have a problem with how it's done. I would much rather do a one second rate-limit. Also, the implementation is way too resource-heavy for what it actually does. The same can be achieved by using a `long` storing the current time and calculating the elapsed time, returning when it's not over one minute.

**Comment by Dimiikou** [comment]
Changed it the way you asked

File: src/main/java/de/fuzzlemann/ucutils/events/ReviveEventHandler.java [hunk 1]
```diff
@@ -11,13 +11,18 @@
...
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
 
 /**
  * @author Fuzzlemann
  */
 @Mod.EventBusSubscriber
 public class ReviveEventHandler {
 
+    private static long latestEcecute = -1;
```
Comment (paulzhng) [inline src/main/java/de/fuzzlemann/ucutils/events/ReviveEventHandler.java:24]: rename to latestExecute

File: src/main/java/de/fuzzlemann/ucutils/events/ReviveEventHandler.java [hunk 2]
```diff
@@ -30,7 +35,11 @@ public static void onRevive(PlayerInteractEvent.RightClickBlock e) {
             double distance = entityItem.getDistanceSq(pos.getX(), pos.getY(), pos.getZ());
 
             if (distance < 2) {
```
Comment (paulzhng) [inline src/main/java/de/fuzzlemann/ucutils/events/ReviveEventHandler.java:37]: please don't nest two if's. Invert this statement & execute return if `distance >= 2`

File: src/main/java/de/fuzzlemann/ucutils/utils/faction/badfaction/speech/SpeechModifyUtil.java [hunk 3]
```diff
@@ -49,10 +49,19 @@ public static String modifyString(String message, SpeechModifier modifier) {
 
     public static String replaceIgnoreCase(String toReplaceString, List<Map.Entry<String, String>> replaceMap) {
         for (Map.Entry<String, String> entry : replaceMap) {
-            String toReplace = entry.getKey();
```
Comment (paulzhng) [inline src/main/java/de/fuzzlemann/ucutils/utils/faction/badfaction/speech/SpeechModifyUtil.java:52]: this doesn't fit in this PR

**Review by paulzhng — CHANGES_REQUESTED** [review]

File: src/main/java/de/fuzzlemann/ucutils/utils/faction/badfaction/speech/SpeechModifyUtil.java (hunk 3 above)
Comment (Dimiikou) [reply src/main/java/de/fuzzlemann/ucutils/utils/faction/badfaction/speech/SpeechModifyUtil.java:52]: > this doesn't fit in this PR

How do i remove this class from this PR?

Comment (paulzhng) [reply src/main/java/de/fuzzlemann/ucutils/utils/faction/badfaction/speech/SpeechModifyUtil.java:52]: Revert it

**Comment by Dimiikou** [comment]
Reverted the Changes, reduced to one if and corrected the missspell

**Comment by paulzhng** [comment]
Changes in SpeechyModifyUtil still aren't reverted

**Comment by Dimiikou** [comment]
But now!

File: src/main/java/de/fuzzlemann/ucutils/events/ReviveEventHandler.java [hunk 4]
```diff
@@ -29,8 +34,9 @@ public static void onRevive(PlayerInteractEvent.RightClickBlock e) {
         for (EntityItem entityItem : items) {
             double distance = entityItem.getDistanceSq(pos.getX(), pos.getY(), pos.getZ());
 
-            if (distance < 2) {
+            if (distance >= 2 && System.currentTimeMillis() - latestExecute < 5000) return;
```
Comment (paulzhng) [inline src/main/java/de/fuzzlemann/ucutils/events/ReviveEventHandler.java:37]: To be honest, I'd lower the cooldown to 500ms or something. Just a QOL change

Comment (Dimiikou) [reply src/main/java/de/fuzzlemann/ucutils/events/ReviveEventHandler.java]: Reduced the cooldown to 500ms

**Review by paulzhng — APPROVED** [review]
LGTM

**Pull request merged**